file_system.go

```diff
@@ -139,6 +139,9 @@ func (c *Client) ReadDir(path string) ([]os.FileInfo, error) {
 			return nil, err
 		}
 		parser = func(entry string, skipSelfParent bool) (os.FileInfo, error) {
+			if c.config.ListParser != nil {
+				return c.config.ListParser.ParseLIST(entry)
```

Comment:
What about the `skipSelfParent` flag? (I think its purpose is to exclude "." and ".." when listing.) Perhaps instead of passing that flag into the parser, we could exclude "." and ".." here using the return value?